File: solved/funnyenc/funny_encryption.py
from sys import stdin
def binary_encrypt(num):
	count = 0
	while(num):
		count += num & 0x01
		num >>= 1
	return count
	
def hex_encrypt(num):
	count = 0
	for d in num: count += binary_encrypt(int(d))
	return count

for s in stdin.read().split('\n')[1:-1]: print "%s %s" % (binary_encrypt(int(s)), hex_encrypt(s))

File: solved/funnyenc/funny_encryption.cpp
#include <iostream>
#include <vector>

using namespace std;

unsigned int binary_encrypt(unsigned int num) {
	unsigned int count = 0;
	do {
		if(num & 0x01) {
			count++;
		}
	} while(num >>= 1);
	return count;
}

unsigned int hex_encrypt(unsigned int num) {
	unsigned int count = 0, dig;
	do {
		dig = num % 10;
		count += binary_encrypt(dig);
	} while(num /= 10);
	return count;
}

int main() {
	unsigned int n, num;
	cin >> n;
	vector<unsigned int> nums;
	while(cin >> num) {
		nums.push_back(num);
	}
	for(int i = 0; i < n; i++) {
		num = nums[i];
		cout << binary_encrypt(num) << " " << hex_encrypt(num) << endl;
	}
	return 0;
}
